habitat/datasets/rearrange/receptacle.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, List, Optional
+
+import magnum as mn
+import numpy as np
+
+import habitat_sim
+
+
+class Receptacle:
```

Comment:
Is the Receptacle class only going to be useful in the rearrange scenario? I think this could be moved outside of datasets maybe?

Reply:
Agreed, this could be useful elsewhere, but no concrete ideas on that yet. Let's defer this to a later refactor.